### PR TITLE
resolvconf: fixes for the compatibility interface 

### LINKED ISSUE #9423: systemd-239 resolvconf functionality never adds nameservers
**systemd version the issue has been seen with**
> 239

**Used distribution**
 > Gentoo

**Expected behaviour you didn't see**
 > `resolvconf -a interface` fails to accept new entries

**Unexpected behaviour you saw**
 > silent failure to add DNS entries via resolvconf, output from a normal run of resolvectl instead.

**Steps to reproduce the problem**
Try running:
`resolvconf -a interface < /path/to/resolv.conf`

### PR DESCRIPTION
**Pull request by yuwata**
Fixes #9423.

### CODE REVIEW

**Comment by yuwata** [comment]
@filbranden @ikelos Could you also test this?

File: src/resolve/resolvconf-compat.c [hunk 1]
```diff
@@ -53,6 +53,8 @@ static int parse_nameserver(const char *string) {
 
                 if (strv_push(&arg_set_dns, word) < 0)
                         return log_oom();
+
+                TAKE_PTR(word);
```
Comment (filbranden) [inline src/resolve/resolvconf-compat.c]: You have a point about `TAKE_PTR()` being unsafe when used as an argument to a function that can fail, but here it's not really necessary, `word = NULL;` is more straightforward, so I'd suggest sticking to that (just fixing the fact that this one didn't set it here.)

Comment (yuwata) [reply src/resolve/resolvconf-compat.c]: Yeah, `word = NULL;` is sufficient. But using `TAKE_PTR()` is not bad. We have already used `TAKE_PTR()` in such cases. C.f. https://github.com/systemd/systemd/blob/ad4bc3352285f467f4ffa03c3171b19fa0a8758d/src/libsystemd/sd-resolve/sd-resolve.c#L1054

Comment (yuwata) [reply src/resolve/resolvconf-compat.c]: I've changed to `word = NULL;`, as the author of the first commit is @filbranden.

**Comment by yuwata** [comment]
Updated. PTAL.

File: src/resolve/resolvconf-compat.c (hunk 1 above)
Comment (filbranden) [reply src/resolve/resolvconf-compat.c]: Thanks!

Yeah, maybe it's OK (the compiler will produce the same) but it's a bit weird seing it used when the value gets discarded, given the whole point of it is preserving the value...

**Pull request merged**